crates/solidity/testing/perf/npm-benches/src/main.rs

```diff
@@ -0,0 +1,192 @@
+        &self,
+        input_path: &Path,
+        projects: &Vec<Project>,
+    ) -> Result<Vec<Timing>> {
+        let mut results = vec![];
+
+        for project in projects {
+            let path = input_path.join(&project.hash);
+
+            let (compiler_version, fully_qualified_name) = Self::compilation_options(&path)?;
+
+            if !regex::Regex::new(&self.pattern)?
+                .is_match(path.join(&fully_qualified_name).to_string_lossy().as_ref())
+            {
+                continue;
+            }
+
+            let mut result =
+                self.run_benchmarks(&compiler_version, &path, &fully_qualified_name)?;
+            results.append(&mut result);
+        }
+        Ok(results)
+    }
+
+    fn compilation_options(path: &Path) -> Result<(String, String)> {
+        let compilation_file = path.join("compilation.json");
+
+        if !compilation_file.exists() {
+            return Err(anyhow::anyhow!(
+                "Missing compilation.json in folder: {:?}",
+                path
+            ));
+        }
+
+        let content = fs::read_to_string(&compilation_file)?;
+        let json: serde_json::Value = serde_json::from_str(&content)?;
+
+        let fully_qualified_name = json
+            .get("fullyQualifiedName")
+            .and_then(|f| f.as_str())
+            .ok_or_else(|| {
+                anyhow::anyhow!("Missing fullyQualifiedName field in file: {compilation_file:?}")
+            })?
+            .rsplit_once(':')
+            .map(|(before_last_colon, _)| before_last_colon)
+            .ok_or_else(|| anyhow::anyhow!("fullyQualifiedName is not well formatted"))?;
+
+        let compiler_version = json
+            .get("compilerVersion")
+            .and_then(|f| f.as_str())
+            .ok_or_else(|| anyhow::anyhow!("Missing compilerVersion in {compilation_file:?}"))?;
+
+        Ok((compiler_version.into(), fully_qualified_name.into()))
+    }
+
+    fn run(
+        &self,
+        compiler_version: &str,
+        path: &Path,
+        fully_qualified_name: &str,
+        runner: Runner,
+    ) -> Result<Vec<Timing>, anyhow::Error> {
+        let command = Command::new("npx")
+            .arg("tsx")
+            .flag("--trace-uncaught")
+            .flag("--expose-gc")
+            .arg("crates/solidity/testing/perf/npm/src/main.mts")
+            .property("--version", compiler_version)
+            .property("--dir", path.to_string_lossy())
+            .property("--file", fully_qualified_name)
+            .property("--runner", runner.to_string())
+            .args(&self.extra_args);
+        let result = command.evaluate()?;
+
+        match serde_json::from_str(&result) {
+            Ok(output) => Ok(output),
+            Err(e) => {
+                eprintln!("Error: Can't parse output as json.\n\tOutput: {result}");
+                Err(e.into())
+            }
+        }
+    }
+}
+
+fn publish(results: &[Timing]) -> Result<()> {
+    let mut output = String::new();
```

Comment:
Nit: If you're only using this to print the results, you can just write directly to stdout (note: you have to import `std::io::Write` for this to work)

```suggestion
    let mut output = std::io::stdout();
```

Then just delete the final `println!()`.

Reply:
ah, I remember why I did this this way: if there's anything printing in the middle, for whatever reason, then  the output will look scrambled. but I don't think that should matter here, so I'll take it.